modules/core/src/main/java/org/locationtech/jts/coverage/TPVWSimplifier.java

```diff
@@ -111,9 +108,11 @@ private List<Edge> createEdges(MultiLineString lines) {
     List<Edge> edges = new ArrayList<Edge>();
     if (lines == null)
       return edges;
+    BitSet freeRings = (lines.getUserData() != null && lines.getUserData() instanceof BitSet) ?
```

Comment:
Not crazy about passing this via userData.  Can't it be a parameter?

And I actually prefer your original idea of passing two sets of lines, one with endpoint constraints and one of the unconstrained rings.

Reply:
Can probably do that, but this separation into two sets of lines is probably of no use for the end user. That's why I thought it would be better to consider it as an implementation detail. I understand the argument to keep things accessible and to avoid hidden tricks though.

Reply:
Ok, either way is fine for now.  If a different calling signature becomes useful it can always be added later.

Reply:
Actually I should have said - the BitSet data is fine, but would be better to pass it as an actual parameter to `TPVWSimplifier`.  I prefer not to hide key data in the untyped `userData` field.